base/abstractarray.jl

```diff
@@ -38,18 +38,12 @@ function first(itr)
 end
 last(a) = a[end]
 
-function stride(a::AbstractArray, i::Integer)
-    if i > ndims(a)
-        return length(a)
-    end
-    s = 1
-    for n=1:(i-1)
-        s *= size(a, n)
-    end
-    return s
-end
+strides{T}(a::AbstractArray{T,0}) = ()
+strides(a::AbstractArray) = _strides(a,(1,))
+_strides{_,N}(a::AbstractArray{_,N}, s::NTuple{N,Int}) = s
+_strides{_,N,M}(a::AbstractArray{_,N}, s::NTuple{M,Int}) = _strides(a, (s..., s[M]*size(a, M)))
 
-strides(a::AbstractArray) = ntuple(i->stride(a,i), ndims(a))::Dims
+stride(a::AbstractArray, d) = d <= 1 ? 1 : size(a, d-1)*stride(a, d-1)
```

Comment:
But that's probably only true if you are probing the last stride, otherwise you are calculating too much. I just don't have that much difference between `f` and `g`, but it is true that `f` (the old definition of `strides`) is slightly faster.